Mathlib/Util/Superscript.lean

```diff
@@ -225,6 +225,30 @@ def scriptParser.formatter (name : String) (m : Mapping) (k : SyntaxNodeKind) (p
   | .ok newStack =>
     set { st with stack := stack ++ newStack }
 
+section Delab
+
+/-- Returns `true` if every character in `s` can be subscripted. -/
+private def isSubscriptable (s : String) : Bool :=
+  s.toList.all fun x ↦ x == ' ' || Mapping.subscript.toSpecial.contains x
+
+/-- Returns `true` if every character in `s` can be superscripted. -/
+private def isSuperscriptable (s : String) : Bool :=
+  s.toList.all fun x ↦ x == ' ' || Mapping.superscript.toSpecial.contains x
+
+private partial def superscriptable : Syntax → DelabM Unit
+  | .node _ _ args => args.forM superscriptable
+  | .atom _ val => guard <| isSuperscriptable val
+  | .ident _ _ val _ => guard <| isSuperscriptable val.toString
+  | _ => failure
+
+private partial def subscriptable : Syntax → DelabM Unit
+  | .node _ _ args => args.forM subscriptable
+  | .atom _ s => guard <| isSubscriptable s
+  | .ident _ _ s _ => guard <| isSubscriptable s.toString
+  | _ => failure
```

Comment:
Can you abstract this to a single function that takes in a `Mapping`. so that it's shared between sub- and super- script?